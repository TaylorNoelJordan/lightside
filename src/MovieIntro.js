import React from 'react'

const MovieIntro = ({ films }) => {
  
  const findEpisode = (specificMovie) => {
      return films.find(movie => movie.episode_id === specificMovie || Math.floor(Math.random() * (7 - 1 + 1)) + 1);
  } 
  
  return (
    <div>
      <p>{findEpisode(4).opening_crawl}</p>
    </div>
  )
}

export default MovieIntro