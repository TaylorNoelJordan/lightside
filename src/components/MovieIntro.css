.btn__hide-splash {
  position: absolute;
  top:10px;
  left:10px;
  z-index: 2;
  border-radius: 50%;
  height:100px;
  width: 100px;
  
}