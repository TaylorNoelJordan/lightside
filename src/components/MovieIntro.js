import React from 'react'
import Crawl from 'react-star-wars-crawl'
import './MovieIntro.css'
import 'react-star-wars-crawl/lib/index.css'

const MovieIntro = ({ films, toggleSplash }) => {
  const {episode_id, title, opening_crawl} = films
  const swapEpisodeNum = () => {
    let epsiodeNumerals = [0, 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII']
    return epsiodeNumerals[episode_id]
  }
  return (
    <article>
      <Crawl
        title={`Episode ${swapEpisodeNum()}`}
        subTitle={title}
        text={opening_crawl}
      />
      <button className="btn__hide-splash" onClick={() => toggleSplash()}>Take me in!</button>
    </article>
  )
}

export default MovieIntro