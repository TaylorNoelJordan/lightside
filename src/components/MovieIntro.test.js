import React from 'react';
import { shallow, mount } from 'enzyme'
import MovieIntro from './MovieIntro';
import results from '../test-data/film-data'

describe('Movie Intro', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MovieIntro toggleSplash={jest.fn()} films={results}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should swap the episode numbers with roman numerals', () => {
    expect(wrapper.instance())
  })

})