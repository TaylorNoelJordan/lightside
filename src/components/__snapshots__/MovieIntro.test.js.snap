// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Intro should match the snapshot 1`] = `
<article>
  <Crawl
    subTitle="A New Hope"
    text="It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy...."
    title="Episode IV"
  />
  <button
    className="btn__hide-splash"
    onClick={[Function]}
  >
    Take me in!
  </button>
</article>
`;
